internal/provider/datasources/work_queue.go

```diff
@@ -0,0 +1,172 @@
+	"github.com/prefecthq/terraform-provider-prefect/internal/provider/customtypes"
+)
+
+var _ = datasource.DataSourceWithConfigure(&WorkQueueDataSource{})
+
+// WorkQueueDataSource contains state for the data source.
+type WorkQueueDataSource struct {
+	client api.PrefectClient
+}
+
+// WorkQueueDataSourceModel defines the Terraform data source model.
+type WorkQueueDataSourceModel struct {
+	ID           customtypes.UUIDValue      `tfsdk:"id"`
+	Created      customtypes.TimestampValue `tfsdk:"created"`
+	Updated      customtypes.TimestampValue `tfsdk:"updated"`
+	AccountID    customtypes.UUIDValue      `tfsdk:"account_id"`
+	WorkspaceID  customtypes.UUIDValue      `tfsdk:"workspace_id"`
+	WorkPoolName types.String               `tfsdk:"work_pool_name"`
+
+	Name             types.String `tfsdk:"name"`
+	Description      types.String `tfsdk:"description"`
+	IsPaused         types.Bool   `tfsdk:"is_paused"`
+	ConcurrencyLimit types.Int64  `tfsdk:"concurrency_limit"`
+	Priority         types.Int64  `tfsdk:"priority"`
+}
+
+// NewWorkQueueDataSource returns a new WorkQueueDataSource.
+//
+//nolint:ireturn // required by Terraform API
+func NewWorkQueueDataSource() datasource.DataSource {
+	return &WorkQueueDataSource{}
+}
+
+// Metadata returns the data source type name.
+func (d *WorkQueueDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_work_queue"
+}
+
+// Configure initializes runtime state for the data source.
+func (d *WorkQueueDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(api.PrefectClient)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected api.PrefectClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	d.client = client
+}
+
+var workQueueAttributes = map[string]schema.Attribute{
+	"id": schema.StringAttribute{
+		Computed:    true,
+		CustomType:  customtypes.UUIDType{},
+		Description: "Work queue ID (UUID)",
+		Optional:    true,
+	},
+	"created": schema.StringAttribute{
+		Computed:    true,
+		CustomType:  customtypes.TimestampType{},
+		Description: "Date and time of the work queue creation in RFC 3339 format",
+	},
+	"updated": schema.StringAttribute{
+		Computed:    true,
+		CustomType:  customtypes.TimestampType{},
+		Description: "Date and time that the work queue was last updated in RFC 3339 format",
+	},
+	"work_pool_name": schema.StringAttribute{
+		Computed:    true,
+		CustomType:  customtypes.TimestampType{},
+		Description: "The name of the work pool that the work queue belongs to",
+	},
+	"name": schema.StringAttribute{
+		Computed:    true,
+		Description: "Name of the work queue",
+		Optional:    true,
+	},
```

Comment:
so when i look at the fetching logic:

```go
client, err := d.client.WorkQueues(model.AccountID.ValueUUID(), model.WorkspaceID.ValueUUID(), model.WorkPoolName.ValueString())
```

^ when we initialize this work queues client, we're looking for three attributes on the datasource:

```hcl
data "prefect_work_queue" "foobar" {
#  ...
  work_pool_name = "my-pool"
  account_id = "ffff-ffff-ffff"
  workspace_id = "ffff-ffff-ffff"
}
```

so first thing is that we need to add `account_id` and `workspace_id` to the attributes map (you have it in the struct already as `AccountID` and `WorkspaceID`)

however, `account_id` and `workspace_id` can be OPTIONAL here, since the `client.WorkQueues()` factory function will fall back to the account_id / workspace_id set on the provider (this is a pattern we've adopted early on - allows the TF operator to create a datasource with an override to a different account/workspace if needed).  Take a look at `datasources/variable.go` as an example

but you will need the `work_pool_name` to be required, i would imagine

Reply:
second, on the actual `Get()` call:

```
queue, err := client.Get(ctx, model.Name.ValueString())
```

^ we need the `name` key to be present, always.  So I think we will want to make this required, instead of optional

for example, on the Get() call, you pass it into the URL path - so we should expect this to always exist when someone creates a work_queue datasource bc it is the main identifier:
https://github.com/PrefectHQ/terraform-provider-prefect/pull/152/files#diff-e146e91f55659265a162378cc118455b6f0a7027ab74c39a8860ebd153dcf5a6R115-R116

Reply:
so if im understanding this correctly, we should always expect, at minimum, these keys to exist on a work_queue datasource - and if so, we need to make them required

```hcl
data "prefect_work_queue" "foobar" {
#  ...
  work_pool_name = "my-pool"
  name = "my-queue"
}
```